compiler/rustc_resolve/src/def_collector.rs

```diff
@@ -138,6 +138,51 @@ impl<'a, 'ra, 'tcx> DefCollector<'a, 'ra, 'tcx> {
         );
         assert!(old_parent.is_none(), "parent `LocalDefId` is reset for an invocation");
     }
+
+    /// Determines whether the const argument's expression is a simple macro call, optionally
+    /// surrounded in braces depending on whether a brace has already been unwrapped on this
+    /// const argument.
+    ///
+    /// If this expresion *is* a trivial macro call then the id for the macro call is
+    /// returned along with the information required to build the anon const's def if
+    /// the macro call expands to a non-trivial expression.
+    fn is_const_arg_trivial_macro_expansion(
+        &self,
+        const_arg_expr: &'a Expr,
+        anon_const: Option<&'a AnonConst>,
+    ) -> Option<(PendingAnonConstInfo, NodeId)> {
+        let (anon_const_id, anon_const_span) = if let Some(anon_const) = anon_const {
+            (anon_const.id, anon_const.value.span)
+        } else if let Some(pending_anon) = &self.pending_anon_const_info {
+            (pending_anon.id, pending_anon.span)
+        } else {
+            panic!(
+                "`is_const_arg_trivial_macro_expansion` called without having recursed into anon const"
+            );
+        };
```

Comment:
Hmm yeah, I originally was gonna have two methods... one that takes an `AnonConst` and one that takes an `Expr` but it wound up being kind of more duplication :thinking: It's probably worth it anyway for clarity since the current setup has real weird function args lol

This kind of logic I don't really want in the caller because it's somewhat subtle and makes `visit_anon_const`/`visit_expr` harder to read.